gitlab-runner-mock/src/api/request.rs

```diff
@@ -60,6 +64,33 @@ impl Respond for JobRequestResponder {
         assert!(r.system_id.starts_with("s_") || r.system_id.starts_with("r_"));
         assert!(r.system_id.len() == 14);
 
+        let expected = self.mock.expected_metadata();
+        if let Some(expected) = expected.version {
+            let version = r.info.version.expect("Missing version in metadata");
+            assert_eq!(expected, version, "Unexpected version in metadata");
```

Comment:
Could these just be `assert_eq!(expected, r.info.version, "...")`? IMO the errors wouldn't be any worse; you can still see the `Some` vs `None` states; if anything it might be a bit clearer because, if the version is missing from the metadata, you'll still get to see the expected one.

Reply:
No because if the mock's expected is None then it's a don't care rather then an error if the request has metadata; I guess i could do:
```rust
if expected.version.is_some() {
  assert_eq!(expected.version, r.info.version, "...")
}
```
but it's not clear to me that's actually a real benefit apart from one less line of code :)

Reply:
that's actually basically what I meant, it's a bit shorter but also:

> if the version is missing from the metadata, you'll still get to see the expected one

that being said that's probably not really a huge deal and I'm also fine with it being left as-is